dsew_community_profile/delphi_dsew_community_profile/pull.py

```diff
@@ -422,6 +422,9 @@ def fetch_listing(params):
     if params['indicator']['reports'] == 'new':
         # drop files we already have in the input cache
         keep = [el for el in listing if not os.path.exists(el['cached_filename'])]
+        if not keep:
+            # No new listings, e.g. on the weekend.
```

Comment:
This same "return all" issue will happen with the `elif params['indicator']['reports'].find("--") > 0:` case too if the range specified returns nothing (e.g. over a weekend).

I think this function has grown and become too confusing. I recommend that we untangle the logic and just return in each of the four branches, even if it comes at the cost of duplicating some code. Here is what reads more clearly to me:
```py
def fetch_listing(params):
    """Generate the list of report files to process."""
    # Can replace epoch start with something like date(1, 1, 1), if you want to be
    # really safe. Can also export to a global var.
    export_start_date = params['indicator'].get('export_start_date', datetime.datetime.fromtimestamp(0))

    listing = requests.get(DOWNLOAD_LISTING).json()['metadata']['attachments']
    # drop the pdf files
    listing = [
        dict(
            el,
            cached_filename=as_cached_filename(params, el),
            publish_date=Dataset.parse_publish_date(el['filename'])
        )
        for el in listing if el['filename'].endswith("xlsx")
    ]

    if params['indicator']['reports'] == 'new':
        # drop files we already have in the input cache
        keep = [el for el in listing if not os.path.exists(el['cached_filename']) and (export_start_date <= el['publish_date'])]
        if not keep:
            # No new listings, e.g. on the weekend.
            return keep
        else:
            return extend_listing_for_interp(keep, listing)
    elif params['indicator']['reports'].find("--") > 0:
        # drop files outside the specified publish-date range
        start_str, _, end_str = params['indicator']['reports'].partition("--")
        start_date = datetime.datetime.strptime(start_str, "%Y-%m-%d").date()
        end_date = datetime.datetime.strptime(end_str, "%Y-%m-%d").date()
        keep = [
            el for el in listing
            if (start_date <= el['publish_date'] <= end_date) and (export_start_date <= el['publish_date'])
        ]
        if not keep:
            # No listing in the selected range, e.g. over a weekend range.
            return keep
        else:
            return extend_listing_for_interp(keep, listing)
    elif params['indicator']['reports'] == 'all':
        keep = [
            el for el in listing if export_start_date <= el['publish_date']
        ]
        return extend_listing_for_interp(keep, listing)
    else:
        raise ValueError(f"params['indicator']['reports'] is set to {params['indicator']['reports']}, which isn't 'new', 'all', or a date range.")
```

Reply:
I'm okay with this so long as we stash 
```
export_start_date <= el['publish_date']
```
in a lambda called in three places instead of copy-pasting it